alea/runner.py

```diff
@@ -164,33 +164,48 @@ def __init__(
             statistical_model_args.get("asymptotic_dof", 1),
         )
 
+    def pre_process_poi(self, value, name):
+        """Pre-process of poi_expectation."""
+        if not all([isinstance(v, (float, int)) for v in value.values()]):
+            raise ValueError(f"{name} should be a dict of float! But {value} is provided.")
+        # update poi according to poi_expectation
+        if "poi_expectation" in value:
+            value = self.update_poi(self.model, self.poi, value, self.nominal_values)
```

Comment:
Since this is the only place where you use `update_poi` why don't you just make it a regular method (non-static) and then only parse `value` here?

Reply:
Maybe users will need it in their notebook.

Reply:
good point!